dask/dataframe/core.py

```diff
@@ -579,6 +579,17 @@ def _aca_agg(self, token, func, aggfunc=None, **kwargs):
         """ Wrapper for aggregations """
         raise NotImplementedError
 
+    def apply_concat_apply(self, func, aggfunc=None, func_kwargs=None,
+                           aggfunc_kwargs=None):
+        """ Apply a function to each partition, the concat, then apply again
+        """
+        token = '-{}-{}'.format('apply_concat_apply',
```

Comment:
`map_partitions` uses something like the following:

```python
    token = kwargs.pop('token', None)
    name = token or funcname(func)
    name = '{0}-{1}'.format(name, tokenize(metadata, *args, **kwargs))
```

I'm afraid that we've been inconsistent about mixing the use of `name` and `token` to mean the same thing.  I agree that the final string should definitely include a tokenized version of the function, and not just its name.  However I also think that using the name of the function as the prefix has a lot of value, particularly if you use any of the diagnostics.